libtransmission/session.c

```diff
@@ -2895,6 +2909,39 @@ void tr_sessionSetQueueEnabled(tr_session* session, tr_direction dir, bool is_en
     session->queueEnabled[dir] = is_enabled;
 }
 
+void tr_sessionSetDownloadDirDynamicEnabled(tr_session* session, bool is_enabled)
+{
+    TR_ASSERT(tr_isSession(session));
+
+    session->downloadDirDynamicEnabled = is_enabled;
+}
+
+bool tr_sessionGetDownloadDirDynamicEnabled(tr_session const* session, tr_direction dir)
```

Comment:
The `tr_direction` argument doesn't seem to be used for anything -- should it be removed?